x/gamm/keeper/grpc_query_test.go

```diff
@@ -87,6 +89,143 @@ func (suite *KeeperTestSuite) TestQueryPools() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestPoolsWithFilter() {
+	var (
+		defaultAcctFunds sdk.Coins = sdk.NewCoins(
+			sdk.NewCoin("uosmo", sdk.NewInt(10000000000)),
+			sdk.NewCoin("foo", sdk.NewInt(10000000)),
+			sdk.NewCoin("bar", sdk.NewInt(10000000)),
+			sdk.NewCoin("baz", sdk.NewInt(10000000)),
+		)
+		defaultPoolParams = balancer.PoolParams{
+			SwapFee: sdk.ZeroDec(),
+			ExitFee: sdk.ZeroDec(),
+		}
+	)
+
+	testCases := []struct {
+		name               string
+		num_pools          int
+		expected_num_pools int
+		min_liquidity      sdk.Coins
+		pool_type          string
+		include_denom      string
+		poolAssets         []balancertypes.PoolAsset
+		expectedErr        error
+	}{
+		{
+			name:               "valid tc",
+			num_pools:          2,
```

Comment:
can we add two test cases for:
- only querying using filter for min liquidity
- only querying using fitler for pool type